controllers/player.controller.js

```diff
@@ -1,43 +1,69 @@
 const Player = require('../models/player.model')
+const logs = require('../helpers/logsHelper')
 
-exports.hello = (req, res) => {
-  res.send('Hello world')
+exports.playerCreate = async (req, res) => {
+  try {
+    const player = await Player.findOneAndUpdate(
+      { name: req.body.name },
+      { $set: { name: req.body.name, score: req.body.score } },
+      { upsert: true }
+    )
+    res.send(player)
+  } catch (error) {
+    const requestId = req.get('kinto-request-id')
+    logs.logError(requestId, error)
+    res.status(400).send({ error: `player not saved: ${error}` })
+  }
 }
 
-exports.player_create = (req, res) => {
-  let player = new Player({
-    name: req.body.name,
-    score: req.body.score
-  })
+exports.playerDetails = async (req, res) => {
+  try {
+    const player = await Player.findOne({ name: req.params.username })
 
-  player.save(function(err) {
-    if (err) {
-      return next(err)
+    if (player === null) {
+      res.status(401).send({ error: 'player not found' })
     }
-    res.send('Player Created successfully')
-  })
-}
 
-exports.player_details = (req, res) => {
-  Player.findById(req.params.id, (err, player) => {
-    if (err) return next(err)
     res.send(player)
-  })
+  } catch (error) {
+    logs.logError(requestId, error)
+    res.status(400).send({ error: `player not found: ${error}` })
```

Comment:
for all the `catch` i would return `500` with unexpected problem happened, because its not the player wasn't found its the database is not working for some reason